EXAMPLES.md

```diff
@@ -119,56 +186,44 @@ On a route that calls an API, check for an expired token and attempt a refresh:
 app.get('/route-that-calls-an-api', async (req, res, next) => {
 
   let apiData = {};
-  let tokenSet = req.openid.tokens;
 
-  if (tokenSet && tokenSet.expired() && tokenSet.refresh_token) {
+  // How the tokenSet is created will depend on how the tokens are stored.
+  let tokenSet = req.openid.makeTokenSet(req.session.openIdTokens);
+  let refreshToken = tokenSet.refresh_token;
+
+  if (tokenSet && tokenSet.expired() && refreshToken) {
     try {
       tokenSet = await req.openid.client.refresh(tokenSet);
     } catch(err) {
       next(err);
     }
 
-    tokenSet.refresh_token = req.openid.tokens.refresh_token;
-    req.openid.tokens = tokenSet;
-  }
+    // New tokenSet may not include a new refresh token.
+    tokenSet.refresh_token = tokenSet.refresh_token ?? refreshToken;
 
-  try {
-    apiData = await request(
-      process.env.API_URL,
-      {
-        headers: { authorization: `Bearer ${tokenSet.access_token}` },
-        json: true
-      }
-    );
-  } catch(err) {
-    next(err);
+    // Where you store the refreshed tokenSet will depend on how the tokens are stored.
+    req.session.openIdTokens = tokenSet;
   }
 
-  res.render('api-data-template', {
-    user: req.openid && req.openid.user,
-    apiData
-  });
+  // Check for and use tokenSet.access_token for the API call ...
 });
 ```
 
-## 5. Calling userinfo
+## 7. Calling userinfo
```

Comment:
Maybe a better title to match the access token one?

Reply:
I struggled with a title here because you only need to call userinfo in specific cases and I'm not sure if I could shorten that down to something universally-clear in the title. I added a bit more color in the description, though.